src/pkg/services/m365/api/lists.go

```diff
@@ -529,6 +551,62 @@ func setAdditionalDataByColumnNames(
 	return filteredData
 }
 
+func hasAddressFields(additionalData map[string]any) (map[string]any, string, bool) {
+	for fieldName, value := range additionalData {
```

Comment:
nit: prefer `k` or `key` when you need a generic name for a map key.
```suggestion
	for k, value := range additionalData {
```